codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpBoundProtocolGenerator.kt

```diff
@@ -667,16 +695,13 @@ private class ServerHttpBoundProtocolTraitImplGenerator(
         Attribute.AllowUnusedMut.render(this)
         rust("let mut input = #T::default();", inputShape.builderSymbol(symbolProvider))
         val parser = structuredDataParser.serverInputParser(operationShape)
+        val noInputs = model.expectShape(operationShape.inputShape).expectTrait<SyntheticInputTrait>().originalId == null
```

Comment:
This is on purpose because of [this test](https://github.com/awslabs/smithy/blob/f8968b9462db03e858b20938c7e82c0c5ac39ac0/smithy-aws-protocol-tests/model/restJson1/empty-input-output.smithy#L199). It's the first example you give, but has a non-empty body and a content-type

Reply:
Ah I see. Interesting. What's the protocol test that asserts that body must be empty when there is no modeled input? I can't seem to find it right now.

Reply:
[This one](https://github.com/awslabs/smithy/blob/603c0f2de0dc54c7ed5cfa0d1e2f1c63d521eed2/smithy-aws-protocol-tests/model/restJson1/malformedRequests/malformed-content-type.smithy#L12) (which corresponds to the second example you give)